staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1990,6 +1990,72 @@ type ClusterTrustBundleProjection struct {
 	Path string `json:"path" protobuf:"bytes,4,rep,name=path"`
 }
 
+// PodCertificateProjection provides a private key and X.509 certificate in the
+// pod filesystem.
+type PodCertificateProjection struct {
+	// Kubelet's generated CSRs will be addressed to this signer.
+	//
+	// +required
+	SignerName string `json:"signerName,omitempty" protobuf:"bytes,1,rep,name=signerName"`
+
+	// The type of keypair Kubelet will generate for the pod.
+	//
+	// Valid values are "RSA3072", "RSA4096", "ECDSAP256", "ECDSAP384", and
+	// "ED25519".
+	//
+	// +required
+	KeyType string `json:"keyType,omitempty" protobuf:"bytes,2,rep,name=keyType"`
+
+	// What is the maximum lifetime of the cert that should be issued?  The
+	// actual issued certificate may have any lifetime *less* than
+	// maxExpirationSeconds.
+	//
+	// This value is copied into the PodCertificateRequest that kubelet creates
+	// to request the certificate.  See the documentation of
+	// PodCertificateRequest.spec.maxExpirationSeconds for information on how
+	// kube-apiserver and admission plugins process this value.
```

Comment:
admission doesn't play a role.  Go ahead and describe what we do with the value inline here.

Reply:
Done.